app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt

```diff
@@ -67,8 +67,17 @@ class ToolbarIntegration(
                                 )
                                 .build()
                         val navController = Navigation.findNavController(toolbar)
-                        if (!navController.popBackStack(R.id.homeFragment, false)) {
-                            navController.navigate(R.id.action_browserFragment_to_homeFragment, null, options, extras)
+                        if (!navController.popBackStack(
+                                R.id.homeFragment,
+                                false
+                            )
+                        ) {
+                            if (navController.currentDestination?.id == R.id.browserFragment) navController.navigate(
+                                R.id.action_browserFragment_to_homeFragment,
+                                null,
+                                options,
+                                extras
+                            ) else navController.navigate(R.id.action_global_home)
```

Comment:
Do we want to do this else here? I think we could just ignore this navigation if we're not on BrowserFragment any longer (ie another destination was navigated to already)

Reply:
Good call. We could just ignore it.